libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/auth/AuthenticationException.kt

```diff
@@ -8,14 +8,21 @@
 package io.element.android.libraries.matrix.impl.auth
 
 import io.element.android.libraries.matrix.api.auth.AuthenticationException
-import org.matrix.rustcomponents.sdk.ClientBuildException as RustAuthenticationException
+import org.matrix.rustcomponents.sdk.ClientBuildException
 
 fun Throwable.mapAuthenticationException(): AuthenticationException {
     val message = this.message ?: "Unknown error"
     return when (this) {
-        is RustAuthenticationException.Generic -> AuthenticationException.Generic(message)
-        is RustAuthenticationException.InvalidServerName -> AuthenticationException.InvalidServerName(message)
-        is RustAuthenticationException.SlidingSyncVersion -> AuthenticationException.SlidingSyncVersion(message)
+        is ClientBuildException -> when (this) {
+            is ClientBuildException.Generic -> AuthenticationException.Generic(message)
+            is ClientBuildException.InvalidServerName -> AuthenticationException.InvalidServerName(message)
+            is ClientBuildException.SlidingSyncVersion -> AuthenticationException.SlidingSyncVersion(message)
```

Comment:
I hope the Client builder retry wasn't affected by this! I think it uses the raw `ClientBuildException.SlidingSyncVersion` since it comes directly from the SDK.

Reply:
Yes, it should be OK.

But just to understand, there is no code behavior change here, this is just about ensuring that all the `ClientBuildException` sub classes are mapped, or am I missing something?

Reply:
No, it's fine. I double checked and we're using the exception from the SDK instead of the mapped one, so even if anything changed here it wouldn't matter.